src/plone/restapi/setuphandlers.py

```diff
@@ -31,19 +26,26 @@ def getNonInstallableProducts(self):  # pragma: no cover
 
 
 def install_pas_plugin(context):
-    uf_parent = aq_inner(context)
-    while True:
-        uf = getToolByName(uf_parent, "acl_users")
-        if IPluggableAuthService.providedBy(uf) and "jwt_auth" not in uf:
+    """
+    Install the JWT token PAS plugin in every PAS acl_users here and above.
+
+    Usually this means it is installed into Plone and into the Zope root.
+    """
+    for uf, is_plone_site in pas.iter_ancestor_pas(context):
+
+        # Add the API token plugin if not already installed at this level
+        if "jwt_auth" not in uf:
             plugin = JWTAuthenticationPlugin("jwt_auth")
             uf._setObject(plugin.getId(), plugin)
             plugin = uf["jwt_auth"]
             plugin.manage_activateInterfaces(
-                ["IAuthenticationPlugin", "IExtractionPlugin"]
+                [
+                    "IAuthenticationPlugin",
+                    "IExtractionPlugin",
+                    "ICredentialsUpdatePlugin",
+                    "ICredentialsResetPlugin",
```

Comment:
Should these plugins really be enabled by default? Isn't this a specific use case? For those who don't care about integrated authentication, wouldn't that be an unnecessary burden (less performance)?

Reply:
> Should these plugins really be enabled by default?

Yes.  The purpose of these changes and this PR is to keep login/logout state in sync between plugins.  Those interfaces are the ones that are responsible for that.

> Isn't this a specific use case?

This is the common use case.  A user who is accessing Plone from multiple interfaces, such as Plone Classic and Volto, and those interfaces use different PAS plugins, such as a cookie plugin and a JWT token plugin, that user has every reason to expect that logging out of one also logs them out of the other.  This is the least surprising behavior and as such should be the default.

> For those who don't care about integrated authentication

We can be grateful to PAS for making that a simple matter of deactivating the plugins for those interfaces in that specific use case.  ;-)

> wouldn't that be an unnecessary burden (less performance)?

Login/logout are rare events and as such not a significant impact on the user experience of performance.  Both of these operations for both of these plugins are also very cheap and as such wouldn't impact performance anyways.  Nor do I know of *any* plugin for which these operations are other than very cheap.

Reply:
Feel free to unresolve with details if this still is blocking merge!

Reply:
Well I'll mark it as unresolved so other people can see your points. But I don't think this is a blocker. I just wish other people would give their opinions too.